sips/sip-tbd/sip-TBD-token-metadata-update-notifications.md

```diff
@@ -0,0 +1,234 @@
+parse its contents, and refresh any contracts that were updated.
+
+# Specification
+
+Notification messages for each token class are specified below. Token metadata update notifications
+must be made via a contract call transaction to the [deployed reference
+contract](https://explorer.stacks.co/txid/0xe92af2ea5c11e2e6fde4d31fd394de888070efff23bffad04465c549543abfa2?chain=mainnet)
+or from a call to `print` within any other contract, including the token contract itself.
+
+The message structure was designed to be reusable by other SIPs who wish to establish new
+notification standards in the future (i.e. by varying the `notification` and `payload` key values).
+
+## Fungible Tokens
+
+When a contract needs to notify the network that metadata has changed for a **Fungible Token**, it
+shall call `print` with a tuple with the following structure:
+
+```clarity
+{ notification: "token-metadata-update", payload: { token-class: "ft", contract-id: <token contract id> }}
+```
+
+| Key                   | Value                                                                  |
+|-----------------------|------------------------------------------------------------------------|
+| `notification`        | The string `"token-metadata-update"`                                   |
+| `payload.token-class` | The string `"ft"`                                                      |
+| `payload.contract-id` | The contract id (principal) of the contract that declared the token    |
+
+## Non-Fungible Tokens
+
+When a contract needs to notify the network that metadata has changed for a **Non-Fungible Token**,
+it shall call `print` with a tuple with the following structure:
+
+```clarity
+{ notification: "token-metadata-update", payload: { token-class: "nft", token-ids: (list u100, u101), contract-id: <token contract id> }}
+```
+
+| Key                   | Value                                                                  |
+|-----------------------|------------------------------------------------------------------------|
+| `notification`        | The string `"token-metadata-update"`                                   |
+| `payload.token-class` | The string `"nft"`                                                     |
+| `payload.token-ids`   | A list with the uint token ids that need to be refreshed               |
+| `payload.contract-id` | The contract id (principal) of the contract that declared the tokens   |
+
+# Backwards compatibility
+
+Developers who need to update metadata for contracts that were deployed before this SIP is activated
+could deploy a new one with a function that broadcasts this message or use the deployed contract
+described in [Reference Implementations](#reference-implementations).
+
+# Activation
+
+This SIP will be activated when at least 10 unique contracts have had metadata updates triggered via
+contract-call transactions that print the proposed notification payload.
```

Comment:
Can you also specify a timeout Stacks block height at which point this SIP will be considered Rejected?  It can be reasonably big -- e.g. 100,000 blocks from now or something.